h2o-bindings/bin/custom/python/gen_infogram.py

```diff
@@ -0,0 +1,230 @@
+        Get the relevance of admissible attributes
+        :param self: 
+        :return: 
+        """
+        if self._model_json["output"]["admissible_relevance"] is None:
+            return None
+        else:
+            return self._model_json["output"]["admissible_relevance"]
+
+    def get_admissible_cmi(self):
+        """
+        Get the normalized cmi of admissible attributes
+        :param self: 
+        :return: 
+        """
+        if self._model_json["output"]["admissible_cmi"] is None:
+            return None
+        else:
+            return self._model_json["output"]["admissible_cmi"]
+
+    def get_admissible_cmi_raw(self):
+        """
+        Get the raw cmi of admissible attributes
+        :param self: 
+        :return: 
+        """
+        if self._model_json["output"]["admissible_cmi_raw"] is None:
+            return None
+        else:
+            return self._model_json["output"]["admissible_cmi_raw"]
+
+    def get_all_predictor_relevance(self):
+        """
+        Get relevance of all predictors
+        :param self: 
+        :return: two tuples, first one is predictor names and second one is relevance
+        """
+        if self._model_json["output"]["all_predictor_names"] is None:
+            return None
+        else:
+            return self._model_json["output"]["all_predictor_names"], self._model_json["output"]["relevance"]
+
+    def get_all_predictor_cmi(self):
+        """
+        Get normalized cmi of all predictors.
+        :param self: 
+        :return: two tuples, first one is predictor names and second one is cmi
+        """
+        if self._model_json["output"]["all_predictor_names"] is None:
+            return None
+        else:
+            return self._model_json["output"]["all_predictor_names"], self._model_json["output"]["cmi"]
+
+    def get_all_predictor_cmi_raw(self):
+        """
+        Get raw cmi of all predictors.
+        :param self: 
+        :return: two tuples, first one is predictor names and second one is cmi
+        """
+        if self._model_json["output"]["all_predictor_names"] is None:
+            return None
+        else:
+            return self._model_json["output"]["all_predictor_names"], self._model_json["output"]["cmi_raw"]
+        
+    # Override train method to support infogram needs
+    def train(self, x=None, y=None, training_frame=None, verbose=False, **kwargs):
+        sup = super(self.__class__, self)
+        
+        def extend_parms(parms):  # add parameter checks specific to infogram
+            if parms["data_fraction"] is not None:
+                assert_is_type(parms["data_fraction"], numeric)
+                assert parms["data_fraction"] > 0 and parms["data_fraction"] <= 1, "data_fraction should exceed 0" \
+                                                                                   " and <= 1."
+        
+        parms = sup._make_parms(x,y,training_frame, extend_parms_fn = extend_parms, **kwargs)
+ 
```

Comment:
This does not work unfortunately.  When user does not set algorithm_params, it is treated as ''.  When user sets algoriothm_params, it is specified as dict.  Hence, during the model preparation: model = h2o.infogram(), it will crash unless the two things in the lower comments are there.  This is all caused by my setting algorithm_params to be string for the client API interface.  However, this will be fixed in the JIRA.